build_examples.cmake

```diff
@@ -0,0 +1,40 @@
+
+option(XMP_WITH_DEMO_PLAYER_SDL1      "Build also demo player for XMP (SDL1 needed)" OFF)
+if(XMP_WITH_DEMO_PLAYER_SDL1)
+    add_executable(xmpsdl
+        examples/player-sdl.c
+    )
+    add_dependencies(xmpsdl XMP_IF)
+
+    find_package(SDL REQUIRED)
+    if(TARGET SDL::SDL)
+        set(SDL_LIBRARY SDL::SDLmain SDL::SDL)
+    endif()
+
+    target_compile_definitions(xmpsdl PRIVATE -DBUILDING_STATIC)
```

Comment:
```suggestion
    if(NOT BUILD_SHARED)
        target_compile_definitions(xmpsdl PRIVATE -DBUILDING_STATIC)
    endif()
```

Reply:
I do think, it's reasonable to add next:
```patch
 if(NOT BUILD_SHARED)
     target_link_libraries(XMP_IF INTERFACE xmp_static)
+     target_compile_definitions(XMP_IF INTERFACE -DBUILDING_STATIC)
 else()
     target_link_libraries(XMP_IF INTERFACE xmp_shared)
 endif()
```
instead of adding this into every other target